test/builders/blocks.go

```diff
@@ -281,3 +290,77 @@ func (c *corruptBlockPair) WithEmptyResultsBlock() *corruptBlockPair {
 	c.rxContainer = &protocol.ResultsBlockContainer{}
 	return c
 }
+
+type blockPairBuilder struct {
+	protocolVersion    primitives.ProtocolVersion
+	virtualChainId     primitives.VirtualChainId
+	currentBlockHeight primitives.BlockHeight
+	tiggerEnabled      bool
+}
+
+func BlockPairBuilder() *blockPairBuilder {
+	return &blockPairBuilder{
+		protocolVersion:    DEFAULT_TEST_PROTOCOL_VERSION,
+		virtualChainId:     DEFAULT_TEST_VIRTUAL_CHAIN_ID,
+		currentBlockHeight: 1000,
+		tiggerEnabled:      true,
+	}
+}
+
+func (b *blockPairBuilder) Build() *protocol.BlockPairContainer {
+	validPrevBlock := BlockPair().WithHeight(b.currentBlockHeight - 1).Build()
+	validPrevBlockHash := digest.CalcTransactionsBlockHash(validPrevBlock.TransactionsBlock)
+	transaction := TransferTransaction().WithProtocolVersion(b.protocolVersion).WithVirtualChainId(b.virtualChainId).WithAmountAndTargetAddress(10, ClientAddressForEd25519SignerForTests(6)).Build()
+	transactionArray := []*protocol.SignedTransaction{transaction}
+	if b.tiggerEnabled {
+		transactionArray = append(transactionArray, TriggerTransaction(b.protocolVersion, b.virtualChainId))
+	}
+	txMetadata := &protocol.TransactionsBlockMetadataBuilder{}
+	txRootHashForValidBlock, _ := digest.CalcTransactionsMerkleRoot(transactionArray)
+	validMetadataHash := digest.CalcTransactionMetaDataHash(txMetadata.Build())
+
+	block := BlockPair().
+		WithHeight(b.currentBlockHeight).
+		WithProtocolVersion(b.protocolVersion).
+		WithVirtualChainId(b.virtualChainId).
+		WithTransactionsArray(transactionArray).
+		WithPrevBlock(validPrevBlock).
+		WithPrevBlockHash(validPrevBlockHash).
+		WithMetadata(txMetadata).
+		WithMetadataHash(validMetadataHash).
+		WithTransactionsRootHash(txRootHashForValidBlock).
+		Build()
+
+	txBlockHashPtr := digest.CalcTransactionsBlockHash(block.TransactionsBlock)
+	receiptMerkleRoot, _ := digest.CalcReceiptsMerkleRoot(block.ResultsBlock.TransactionReceipts)
+	stateDiffHash, _ := digest.CalcStateDiffHash(block.ResultsBlock.ContractStateDiffs)
+	preExecutionRootHash := &services.GetStateHashOutput{
+		StateMerkleRootHash: hash.CalcSha256([]byte{1, 2, 3, 4, 5}),
+	}
+
+	block.ResultsBlock.Header.MutateTransactionsBlockHashPtr(txBlockHashPtr)
+	block.ResultsBlock.Header.MutateReceiptsMerkleRootHash(receiptMerkleRoot)
+	block.ResultsBlock.Header.MutateStateDiffHash(stateDiffHash)
+	block.ResultsBlock.Header.MutatePreExecutionStateMerkleRootHash(preExecutionRootHash.StateMerkleRootHash)
+
+	return block
+}
+
+func (b *blockPairBuilder) WithCfg(cfg config.ConsensusContextConfig) *blockPairBuilder {
```

Comment:
please use a locally-defined interface here to avoid coupling to `ConsensusContextConfig`